cli/js/compiler.ts

```diff
@@ -71,9 +74,29 @@ interface CompileResult {
   diagnostics?: Diagnostic;
 }
 
+// this is used to generate the foundational AST for all other compilations, so
+// it can be cached as part of the
+let oldProgram: ts.Program;
+
+((): void => {
+  const ops = core.ops();
+  for (const [name, opId] of Object.entries(ops)) {
+    const opName = `OP_${name.toUpperCase()}` as keyof typeof dispatch;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (dispatch as any)[opName] = opId;
+  }
+  const host = new Host({ writeFile(): void {} });
+  const options = host.getCompilationSettings();
```

Comment:
@kitsonk could you mark this bit as TODO/FIXME for later?

Reply:
This line, in what sense?

Reply:
Ooops, not that line, rather the `for ... of`